chainer/training/extensions/snapshot_writers.py

```diff
@@ -57,6 +57,28 @@ def save(self, filename, outdir, target, savefun, **kwds):
             savefun(tmppath, target)
             shutil.move(tmppath, os.path.join(outdir, filename))
 
+        self._post_save()
+
+    def _add_cleanup_hook(self, hook_fun):
+        """Adds cleanup hook function.
+
+        Technically, arbitrary user-defined hook can be called, but
+        this is intended for cleaning up stale snapshots.
+
+        Args:
+            hook_fun (callable): callable funtion to be called
+                right after save is done. It takes no arguments.
+
+        """
+        if not hasattr(self, '_post_save_hooks'):
+            self._post_save_hooks = []
+        self._post_save_hooks.append(hook_fun)
+
+    def _post_save(self):
+        if hasattr(self, '_post_save_hooks'):
```

Comment:
By applying the comment above, this check can be omitted.

Reply:
Removed; thanks